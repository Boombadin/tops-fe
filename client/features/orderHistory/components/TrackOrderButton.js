import { breakpoint } from '@central-tech/core-ui';
import React from 'react';
import styled from 'styled-components';

import withLocales from '@client/hoc/withLocales';
import {
  ELEMENT_ACTION,
  ELEMENT_TYPE,
  generateTestId,
} from '@client/utils/generateElementId';

const OrderTracking = styled.button`
  width: 100px;
  height: 32px;
  font-size: 13px;
  align-self: center;
  min-width: 80px;
  max-width: 120px;
  padding: 5px 0;
  line-height: 22px;
  font-weight: bold;
  border-radius: 5px;
  border: solid 1px #ed1d23;
  background-color: #ed1d23;
  color: #ffffff;
  cursor: pointer;
  margin: 0 8px 0 27px;
  ${breakpoint('xs', 'md')`
    width: 100%;
    height: 36px;
    max-width: none;
    margin: 0 0 5px 0;
  `}
`;

const TrackOrderButton = ({ translate, orderId, handleTrackClick }) => {
  return (
    <OrderTracking
      data-testid={generateTestId({
        type: ELEMENT_TYPE.BUTTON,
        action: ELEMENT_ACTION.VIEW,
        moduleName: 'TrackOrderButton',
        uniqueId: `handleTrackClick-${orderId}`,
      })}
      onClick={handleTrackClick}
      color="#544943"
      radius={5}
    >
      {translate('order_history.table.track_order')}
    </OrderTracking>
  );
};

export default withLocales(TrackOrderButton);
