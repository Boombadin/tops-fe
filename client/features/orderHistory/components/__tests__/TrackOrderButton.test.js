import { fireEvent, render } from '@testing-library/react';
import React from 'react';

import TrackOrderButton from '../TrackOrderButton';

jest.mock('@client/hoc/withLocales', () => {
  return Component => props => (
    <Component translate={jest.fn(key => key)} {...props} />
  );
});

const TEST_ID = {
  BUTTON_TRACKORDER: 'btn-viewTrackOrderButton-handleTrackClick-621636',
};

describe('features/orderDetail/components/TrackOrderButton', () => {
  const props = {
    orderId: '621636',
    handleTrackClick: () => {},
  };

  test('it should be match snapshot', () => {
    const { asFragment } = render(<TrackOrderButton {...props} />);
    expect(asFragment(<TrackOrderButton {...props} />)).toMatchSnapshot();
  });

  test('click open tracking process from grap | it should be match snapshot', () => {
    const { getByTestId, asFragment } = render(<TrackOrderButton {...props} />);
    const buttonViewOrderTracking = getByTestId(TEST_ID.BUTTON_TRACKORDER);
    expect(buttonViewOrderTracking).toBeInTheDocument();
    fireEvent.click(buttonViewOrderTracking);
    expect(asFragment()).toMatchSnapshot();
  });
});
