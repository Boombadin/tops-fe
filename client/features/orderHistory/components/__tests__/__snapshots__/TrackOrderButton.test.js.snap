// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features/orderDetail/components/TrackOrderButton click open tracking process from grap | it should be match snapshot 1`] = `
<DocumentFragment>
  <button
    class="TrackOrderButton__OrderTracking-sziof4-0 jpGFYg"
    color="#544943"
    data-testid="btn-viewTrackOrderButton-handleTrackClick-621636"
    radius="5"
  >
    order_history.table.track_order
  </button>
</DocumentFragment>
`;

exports[`features/orderDetail/components/TrackOrderButton it should be match snapshot 1`] = `
<DocumentFragment>
  <button
    class="TrackOrderButton__OrderTracking-sziof4-0 jpGFYg"
    color="#544943"
    data-testid="btn-viewTrackOrderButton-handleTrackClick-621636"
    radius="5"
  >
    order_history.table.track_order
  </button>
</DocumentFragment>
`;
